tests/phpunit/tests/option/option.php

```diff
@@ -255,4 +255,24 @@ public function data_option_autoloading() {
 			array( 'autoload_false', false, 'no' ),
 		);
 	}
+
+	/**
+	 * Check we can shortcut the get_option with the pre_option_all filter.
+	 *
+	 * @ticket 37930
+	 *
+	 * @covers get_option
+	 */
+	public function test_filter_pre_option_all_filter_is_called() {
+
+		add_filter( 'pre_option_all', array( $this, 'pre_option_all_filter' ) );
+
+		// Filter was not called.
+		$this->assertEquals( array( 'filtered' => 'true' ), get_option( 'ignored' ) );
+	}
+
+	public function pre_option_all_filter() {
+		return array( 'filtered' => 'true' );
```

Comment:
The test suite has a utility class to ensure filters are called.

Note: I've used my suggested name change here

```suggestion
		$filter = new MockAction();
		add_filter( 'pre_option', array( $filter, 'filter' ) );
		get_option( 'ignored' );

		// Filter was called.
		$this->assertSame( 1, $a->get_call_count() );
```